docs/stopping_events/syscalls.md

```diff
@@ -0,0 +1,206 @@
+
+## :material-arrow-decision: Hijacking
+When hijacking a syscall, the user can provide an alternative syscall to be executed in place of the original one. Internally, the hijack is implemented by registering a handler for the syscall and replacing the syscall number with the new one.
+
+!!! ABSTRACT "Function Signature"
+    ```python
+    d.hijack_syscall(original_syscall, new_syscall, recursive=False, **kwargs) 
+    ```
+
+**Parameters**:
+
+| Argument | Type | Description |
+| --- | --- | --- |
+| `original_syscall` | `int` \| `str` | The syscall number or name to be hijacked. If set to `"*"` or `"all"`, all syscalls will be hijacked. |
+| `new_syscall` | `int` \| `str` | The syscall number or name to be executed instead. |
+| `recursive` | `bool` | If set to `True`, the handler's callback will be executed even if the syscall was triggered by a hijack. |
+| `**kwargs` | `(int, optional)` | Additional arguments to be passed to the new syscall. |
+
+**Returns**:
+
+| Return | Type | Description |
+| --- | --- | --- |
+| `SyscallHandler` | [SyscallHandler](../../from_pydoc/generated/data/syscall_handler) | The handler object created. |
+
+
+!!! ABSTRACT "Example of hijacking a syscall"
+    <div class="grid cards" markdown>
+
+    ```C
+    #include <unistd.h>
+
+    char secretBuffer[32] = "The password is 12345678";
+
+    int main(int argc, char** argv)
+    {
+        [...]
+
+        read(0, secretBuffer, 31);
+        
+        [...]
+        return 0;
+    }
+    ```
+
+    ```python
+    from libdebug import debugger
+
+    d = debugger("./test_program")
+    d.run()
+
+    handler = d.hijack_syscall("read", "write")
+    
+    d.cont()
+    d.wait()
+
+    out = pipe.recvline()
+    print(out.decode())
+    ```
+
+    </div>
+    In this case, the secret will be leaked to the standard output instead of being overwritten with content from the standard input.
+
+For your convenience, you can also easily provide the syscall parameters to be used when the hijacked syscall is executed:
+
+!!! ABSTRACT "Example of hijacking a syscall with parameters"
+    <div class="grid cards" markdown>
+
+    ```C
+    #include <unistd.h>
+
+    char manufacturerName[32] = "libdebug";
+    char secretKey[32] = "provola";
+
+    int main(int argc, char** argv)
+    {
+        [...]
+
+        read(0, manufacturerName, 31);
+        
+        [...]
+        return 0;
+    }
+    ```
+
+    ```python
+    from libdebug import debugger
+
+    d = debugger("./test_program")
+    d.run()
+
+    manufacturerBuffer = [...]
+
+    handler = d.hijack_syscall("read", "write",
+        syscall_arg0=0x1,
+        syscall_arg1=manufacturerBuffer,
+        syscall_arg2=0x100
+    )
+        
+    d.cont()
+    d.wait()
+
+    out = pipe.recvline()
+    print(out.decode())
+    ```
+
+    </div>
+
+    Again, the secret will be leaked to the standard output.
```

Comment:
This is now leaking the manufacturer I believe, not the secret

Reply:
This leaks both due to the specified size of the write

Reply:
Ah my bad, you're right